src/routes/add-article.ts

```diff
@@ -1,16 +1,24 @@
 import createHttpError from 'http-errors';
 import { constants } from 'http2';
-import { JsonLdObj } from 'jsonld/jsonld-spec';
+import jsonld from 'jsonld';
+import { JsonLdArray } from 'jsonld/jsonld-spec';
 import { Context, Middleware, Next } from 'koa';
 import uniqueString from 'unique-string';
+import { schema } from 'rdf-namespaces';
 import Articles from '../articles';
 
 export default (articles: Articles): Middleware => (
   async ({ request, response }: Context, next: Next): Promise<void> => {
-    const article = request.body as JsonLdObj;
+    const [article] = await jsonld.expand(request.body) as JsonLdArray;
+    if (!(article['@type'].includes(schema.Article))) {
+      throw new createHttpError.Forbidden(`Article type must be ${schema.Article} ('${article['@type'].join(', ')}' was given)`);
+    }
     if ('@id' in article) {
       throw new createHttpError.Forbidden(`Article IDs must not be set ('${article['@id']}' was given)`);
     }
+    if (!(schema.name in article) || article[schema.name].length === 0) {
+      throw new createHttpError.Forbidden(`Article must have at least one ${schema.name}`);
+    }
```

Comment:
the checks seem functionally correct and consistent with existing ones like the one on `@id`